bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -257,12 +257,15 @@ public static boolean anyToJBoolean(Object sourceVal) {
      * @return true if the value belongs to the given type, false otherwise
      */
     public static boolean checkIsType(Object sourceVal, Type targetType) {
-        Context cx = context();
         Type sourceType = getType(sourceVal);
+        if (sourceType == targetType || (sourceType.getTag() == targetType.getTag() && sourceType.equals(targetType))) {
```

Comment:
I think with this change we no longer this equals check in TypeCheckCache https://github.com/ballerina-platform/ballerina-lang/blob/c4fe578f0af44a71df17d439d4fe0999b6bee248/bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/types/semtype/TypeCheckCache.java#L32